File: ios/Khrysalis Template/com/lightningkite/khrysalistemplate/vg/MarginTestVG.shared.swift
//Package: com.lightningkite.khrysalistemplate.vg
//Converted using Khrysalis2

import Foundation
import Khrysalis
import RxSwift
import RxRelay
import KhrysalisMaps
import KhrysalisBluetooth



public class MarginTestsVG: ViewGenerator {
    
    
    override public var title: String {
        get {
            return "Margin Tests"
        }
    }
    
    override public func generate(dependency: ViewDependency) -> View {
        var xml = MarginTestXml()
        var view = xml.setup(dependency)
        xml.scrollToTop.onClick(captureWeak(xml.scrollView) { (scrollView) in 
            scrollView.smoothScrollTo(0, 0)
        })
        return view
    }
    override public func generate(_ dependency: ViewDependency) -> View {
        return generate(dependency: dependency)
    }
    
    override public init() {
        super.init()
    }
}
 

File: android/src/main/java/com/lightningkite/khrysalistemplate/vg/MarginTestVG.shared.kt
package com.lightningkite.khrysalistemplate.vg

import android.view.View
import com.lightningkite.khrysalis.captureWeak
import com.lightningkite.khrysalis.views.ViewDependency
import com.lightningkite.khrysalis.views.ViewGenerator
import com.lightningkite.khrysalis.views.onClick
import com.lightningkite.khrysalistemplate.layouts.MarginTestXml

class MarginTestsVG : ViewGenerator() {
    override val title: String get() = "Margin Tests"

    override fun generate(dependency: ViewDependency): View {
        val xml = MarginTestXml()
        val view = xml.setup(dependency)
        xml.scrollToTop.onClick {
            xml.scrollView.smoothScrollTo(0,0)
        }
        return view
    }
}
